File: Games/NoughtsAndCrosses/src/main/java/jokatu/game/games/noughtsandcrosses/input/NoughtsAndCrossesInputDeserialiser.java
package jokatu.game.games.noughtsandcrosses.input;

import jokatu.game.input.DeserialisationException;
import jokatu.game.input.InputDeserialiser;
import org.jetbrains.annotations.NotNull;
import org.springframework.stereotype.Component;

import java.util.Map;

@Component
public class NoughtsAndCrossesInputDeserialiser implements InputDeserialiser<NoughtsAndCrossesInput> {
	@NotNull
	@Override
	public NoughtsAndCrossesInput deserialise(@NotNull Map<String, Object> json) throws DeserialisationException {
		return new NoughtsAndCrossesInput((Integer) json.get("choice"));
	}
}

File: Games/NoughtsAndCrosses/src/main/kotlin/jokatu/game/games/noughtsandcrosses/input/NoughtsAndCrossesInputDeserialiser.kt
package jokatu.game.games.noughtsandcrosses.input

import jokatu.game.input.DeserialisationException
import jokatu.game.input.TypedSingleKeyInputDeserialiser
import org.springframework.stereotype.Component

@Component
class NoughtsAndCrossesInputDeserialiser : TypedSingleKeyInputDeserialiser<Int, NoughtsAndCrossesInput>(Int::class) {

	override val keyName: String
		get() = "choice"

	@Throws(DeserialisationException::class)
	override fun deserialiseTypedSingleValue(json: Map<String, Any>, choice: Int): NoughtsAndCrossesInput {
		return NoughtsAndCrossesInput(choice)
	}
}
